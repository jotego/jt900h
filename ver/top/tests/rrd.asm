    ; RRD
    maxmode on
    relaxed on
    org 0
    ld a,0xbf    ; common header

    ld xhl,data
    ld a,0
    ld bc,8

scroll:
    rrd a,(xhl+)
    djnz bc,scroll

    cp bc,0
    jp ne,bad_end

    ld xhl,data
    ld xix,ref
    ld bc,8
ver:
    ld a,(xhl+)
    cp a,(xix+)
    jp ne,bad_end
    djnz bc,ver

end_loop:
    ldf 0
    ld hl,0xbabe
    ld (0xffff),0xff
    jp end_loop
bad_end:
    ld hl,0xdead
    ld (0xffff),0xff
    jp bad_end
data:
    db 0x12,0x34,0x56,0x78,0x9A,0xBC,0xDE,0xF0
    dw 0xbbbb,0x0000,0x0000,0x0000
ref:
    db 0x01,0x23,0x45,0x67,0x89,0xAB,0xCD,0xEF
    dw 0x0000,0x0000,0x0000,0x0000
    dw 0x0000,0x0000,0x0000,0x0000,0x0000,0x0000,0x0000,0x0000
copy:
    dw 0x0000,0x0000,0x0000,0x0000,0x0000,0x0000,0x0000,0x0000
    end