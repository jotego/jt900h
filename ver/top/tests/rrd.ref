000000ef - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
00000064 - 00000000 - 00000000 - 00000100
SR = f842
