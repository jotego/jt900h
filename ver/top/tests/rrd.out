000000ef - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
0000005c - 00000000 - 00000000 - 00000000
SR = f842
